test/test_ops.py

```diff
@@ -113,19 +113,27 @@ def variant_out_fn(*args, **kwargs):
                 variant_out_fn = variant
 
             def fn(*inputs):
+                # Pack input back into TensorList since we splat it when passing to gradcheck
+                if not isinstance(sample.input, torch.Tensor):
+                    n = len(sample.input)
+                    inputs = (inputs[:n], *inputs[n:])
```

Comment:
out of curiosity, why do you to `inputs[:n]` and `*inputs[n:]` here? I'm not familiar with the gradcheck code but this looks odd

Reply:
gradcheck does not accept TensorList, so when we call it from test_ops we splat TensorList inputs into just n single Tensor inputs. When calling the op we need to put them back together into a TensorList. That is what this code is doing.

Reply:
- [ ] revise the comments to make this clearer